src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) storage.save(addressBook);
```

Comment:
Code style violation. Check how you should write a if statement in Java Coding Standard for this module.